### DISCUSSION TITLE
Support for GraphQL subscriptions

### DISCUSSION THREAD

**Discussion by imetallica** [fastapi/fastapi#7875] [Questions] [answered]
Hi, is it possible to make it work with [graphql-ws](https://github.com/graphql-python/graphql-ws) so we can have subscriptions work out of the box?

Thanks for the awesome library!

I.

**Comment by tiangolo** [comment, marked as answer]
They seem to support Django Channels: https://github.com/graphql-python/graphql-ws#django-channels

Django Channels is ASGI, so I would assume that you could adapt it and use that same version for Django Channels here.

I think this would be a valid feature request for that repo (in an issue in that repo). It would be to "Add support for ASGI frameworks".

And they probably already have support for ASGI, but they assume it's only Django Channels... they might not know that they (probably) already implemented the ASGI standard while implementing support for Django Channels.

I'll close this issue, as the feature request would actually be in that package, not here. But feel free to add more comments or issues.
